src/base/bittorrent/torrenthandle.cpp

```diff
@@ -140,11 +146,11 @@ CreateTorrentParams::CreateTorrentParams(const AddTorrentParams &params)
 const qreal TorrentHandle::USE_GLOBAL_RATIO = -2.;
 const qreal TorrentHandle::NO_RATIO_LIMIT = -1.;
 
-const int TorrentHandle::USE_GLOBAL_SEEDING_TIME = -2;
-const int TorrentHandle::NO_SEEDING_TIME_LIMIT = -1;
+const std::chrono::minutes TorrentHandle::USE_GLOBAL_SEEDING_TIME {-2};
```

Comment:
What `duration` are you talking about?
`std::chrono::duration` is just template and `std::chrono::minutes` (etc.) is its instantiation.
Or am I missing something?

Reply:
> What duration are you talking about?

clock_type::duration typedef.

Reply:
>type safety and not having time unit bugs

I'm sorry, but I'm too stupid and I don't understand how this affects qBittorrent code outside of the libtorrent abstraction layer. All we need there (except the interaction with libtorrent abstraction layer) is to get some duration value from the user through some control providing us with a raw integer, display this value on the screen through some control getting from us a raw integer.

Reply:
I've seen lot's of code that does other things to durations too, comparing them for instance.
What type safety buys you is to not have to name functions and variables with "InMinutes" or "InSeconds". It lowers cognitive load of the reader as there is no longer a concern of what unit a raw integer is.

Reply:
Maybe I don't like it at the moment because the new code isn't clean enough yet. 
Really I like the pros of these types, but the code changes look like "solid inconveniences." Although I've watched some not all. Maybe the whole picture isn't so scary.

As of using concrete types I agree with @arvidn. We should use "duration" everywhere unless other type is required. I.e. function types and parameters, class fields and constants should be of type "duration" (we should add alias for it, e.g. "BitTorrent::Duration").